x-pack/plugins/reporting/public/lib/stateless_status_context.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { createContext, useContext, FunctionComponent } from 'react';
+import { DoNotCheckIlmPolicyStatus } from './reporting_api_client';
+
+type DoNotCheckIlmPolicy = ReturnType<typeof DoNotCheckIlmPolicyStatus>;
+
+interface ContextValue {
+  recheckStatus: DoNotCheckIlmPolicy['resendRequest'];
+}
+
+const PolicyStatusContext = createContext<undefined | ContextValue>(undefined);
+
+export const PolicyStatusContextProvider: FunctionComponent = ({ children }) => {
+  const { resendRequest: recheckStatus } = DoNotCheckIlmPolicyStatus();
+
+  return (
+    <PolicyStatusContext.Provider value={{ recheckStatus }}>
+      {children}
+    </PolicyStatusContext.Provider>
+  );
+};
+
+export type DoNotUseIlmPolicyStatusReturn = ReturnType<typeof DoNotUseIlmPolicyStatus>;
+
+export const DoNotUseIlmPolicyStatus = (): ContextValue => {
```

Comment:
```suggestion
export type UseDefaultPolicyStatusReturn = ReturnType<typeof useDefaultPolicyStatus>;

export const useDefaultPolicyStatus = (): ContextValue => {
```

Reply:
Hm, these actually never really need to be used: in `report_listing.tsx`, when statefulSettings are not enabled, we can render `ReportListingUi` without any prop for context value. https://github.com/elastic/kibana/pull/166493/files#r1379390495